app/src/main/kotlin/com/metrolist/music/utils/potoken/PoTokenGenerator.kt

```
@@ -24,49 +24,61 @@ class PoTokenGenerator {
+                val result = runBlocking {
+                    withTimeout(POTOKEN_TIMEOUT_MS) {
+                        getWebClientPoToken(videoId, sessionId, forceRecreate = attempt > 0)
                     }
-                    webPoTokenGenerator = null
-                    webPoTokenStreamingPot = null
-                    webPoTokenSessionId = null
                 }
+                if (result != null) return result
+                Timber.tag(TAG).w("PoToken returned null on attempt ${attempt + 1}; not retrying")
+                return null
+            } catch (e: TimeoutCancellationException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)